lib/spack/spack/operating_systems/mac_os.py

```diff
@@ -143,6 +143,7 @@ def __init__(self):
             "12": "monterey",
             "13": "ventura",
             "14": "sonoma",
+            "15": "sequoia",
```

Comment:
This also needs some change in `os_compatible.lp` (to be able to use binaries from previous os versions with `sequioia`)